### PR TITLE
toastify: init at 0.5.2

### PR DESCRIPTION
**Pull request by colemickens**
###### Description of changes

new rust utility to send desktop notifications

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/toastify/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchFromGitHub {
+    owner = "hoodie";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-fCwxFdpwtG83xw3DDt9rlnbY8V3eKemRFK/6E1Bhm4c=";
+  };
+
+  cargoHash = "sha256-ecc3z0T82pYR9gSYZYxRYhse9IroydPOAtRgDWqHTbo=";
+
```
Comment (figsoda) [inline pkgs/tools/misc/toastify/default.nix:15]:
```suggestion

  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk_11_0.frameworks.Cocoa
  ];

  preBuild = lib.optionalString stdenv.isDarwin ''
    export HOME=$(mktemp -d)
  '';

```

File: pkgs/tools/misc/toastify/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ lib, rustPlatform, fetchFromGitHub }:
```
Comment (figsoda) [inline pkgs/tools/misc/toastify/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, stdenv, darwin }:
```

File: pkgs/tools/misc/toastify/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-fCwxFdpwtG83xw3DDt9rlnbY8V3eKemRFK/6E1Bhm4c=";
+  };
+
+  cargoHash = "sha256-ecc3z0T82pYR9gSYZYxRYhse9IroydPOAtRgDWqHTbo=";
+
+  meta = with lib; {
+    description = "A commandline tool that shows desktop notifications using notify-rust";
+    homepage = "https://github.com/hoodie/toastify";
```
Comment (figsoda) [inline pkgs/tools/misc/toastify/default.nix:18]:
```suggestion
    homepage = "https://github.com/hoodie/toastify";
    changelog = "https://github.com/hoodie/toastify/blob/${src.rev}/CHANGELOG.md";
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -10746,6 +10746,8 @@ with pkgs;
 
   tinystatus = callPackage ../tools/networking/tinystatus { };
 
+  toastify = callPackage ../tools/misc/toastify {};
```
Comment (figsoda) [inline pkgs/top-level/all-packages.nix:10749]:
```suggestion
  toastify = darwin.apple_sdk_11_0.callPackage ../tools/misc/toastify {};
```

**Review by figsoda — COMMENTED** [review]
I was able to build this on {aarch64,x86_64}-darwin with this patch

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (colemickens) [reply pkgs/top-level/all-packages.nix:10749]: had to do this manually, it kept telling me the diff changed on webui even after refreshing

**Comment by colemickens** [comment]
fixed up, rebuilt/tested, pushed. thanks for the review and darwin compat!

**Review by figsoda — APPROVED** [review]

**Pull request merged**